dask/dataframe/tests/test_dataframe.py

```diff
@@ -4135,6 +4135,24 @@ def test_set_index_with_index():
     assert ddf2 is ddf
 
 
+def test_set_index_no_sort():
+    """
+    GH10333 - Allow setting index on existing partitions without
+    computing new divisions and repartitioning.
+    """
+    df = dd.from_dict({"col1": [2, 4, 1, 3, 5], "col2": [1, 2, 3, 4, 5]}, npartitions=2)
+
+    # Default is sort=True
+    result = df.set_index("col1")
+    assert result.divisions == (1, 2, 5)
+    assert result.index.compute().tolist() == [1, 2, 3, 4, 5]
```

Comment:
For default (`sort=True`) I kept the explicit test of an ordered result; otherwise it seemed indirectly testing the same thing (values same but different order) was an unnecessary test indirection; but the `assert_eq` suggestion here worked well for the second portion (`sort=False`). Thanks. https://github.com/dask/dask/pull/10342/commits/beb40dcd9f4157f35eafdb4b0e436a909f3295c2